src/macfloat.c

```diff
@@ -70,6 +71,16 @@ static void PrintMacfloatToBuf(char *buf, size_t bufsize, Double val, int precis
     }
     else {
         snprintf(buf, bufsize, "%.*" PRINTFFORMAT, precision, val);
+        if (strlen(buf) < bufsize) {
+            char * loc = strchr(buf, 'e');
+            if (loc && !strchr(buf, '.')) {
+                memmove(loc + 1, loc, strlen(loc) + 1);
+                loc[0] = '.';
+            }
+            else if (!strchr(buf, '.')) {
```

Comment:
```suggestion
            else {
```

Reply:
No, that if is correct. The string might be `1.7`

Reply:
But then we never get here, because we already check for the `.` a few lines above (with my suggested change)

Reply:
Argh, I see now, okay :). I was confused by the long change, and thought it was a comment.